configuration/debian/tedge_agent/postinst

```diff
@@ -2,12 +2,18 @@
 
 set -e
 
-# Initialize the agent __only__ if not when agent is running
+# Initialize the agent __only__ if not when agent is running.
 # Indeed, during an OTA self-update, the previous version of the  `agent` is kept running
 # to monitor the installation of the new version up to completion.
 if command -v systemctl >/dev/null; then
-    if ! systemctl is-active --quiet tedge-agent; then
-         sudo -- tedge_agent --init
+    if ! systemctl is-active --quiet tedge-agent ; then
+         # Flock file should be removed as `tedge_agent --init` must run as root
+         # although the flockfile is owned by `tedge`.
+         if [ -f "/run/lock/tedge_agent.lock" ]; then
+                rm /run/lock/tedge_agent.lock
+         fi
+         # It must be run as root to create the directory `/var/tedge`.
+         tedge_agent --init
```

Comment:
But then, we will have a lock own by root, preventing the agent to run as tedge. Am I missing something?

Reply:
My experiment on my laptop, `tedge_agent.lock` is gone after `tedge_agent --init`. However, to keep it safe, I can add the lines to remove `/run/lock/tedge_agent.lock` after `tedge_agent --init`.
```shell
         if [ -f "/run/lock/tedge_agent.lock" ]; then
                rm /run/lock/tedge_agent.lock
         fi
```

This was the output when I ran `sudo tedge_agent --init`.
```
rina@SAG-CFPPNV2-U:/etc/tedge$ inotifywait -m /run/lock/
Setting up watches.
Watches established.
/run/lock/ CREATE tedge_agent.lock
/run/lock/ OPEN tedge_agent.lock
/run/lock/ CLOSE_WRITE,CLOSE tedge_agent.lock
/run/lock/ DELETE tedge_agent.lock
```

```
rina@SAG-CFPPNV2-U:~$ sudo tedge_agent --init
[sudo] password for rina: 
2022-11-04T13:18:47.710656337Z  INFO flockfile::unix: Lockfile created "/run/lock/tedge_agent.lock"
2022-11-04T13:18:47.710699697Z  INFO tedge_agent::agent: tedge_agent starting
2022-11-04T13:18:47.712704417Z  INFO sm-agent{config_dir="/etc/tedge"}: tedge_agent::agent: Initializing the tedge agent session
```

Reply:
Okay. It makes sense and it's useless to remove the lock after --init.